src/internal/components/chart-popover/index.tsx

```diff
@@ -77,6 +78,12 @@ export default function ChartPopover({
     };
   }, [container, onDismiss]);
 
+  const dismissButtonRefCallback = useCallback((element: ButtonProps.Ref) => {
+    if (element) {
+      element.focus();
```

Comment:
why is this necessary?

Reply:
To focus the dismiss button when the popover is already open, but is changed into "locked" mode by clicking/pressing enter on a specific point in the chart. The focus-lock library seemed to handle this magically somehow, but doing it more explicitly seems to make sense. Perhaps it would make sense to move this logic elsewhere though, I'll look into this

Reply:
I've looked, and don't think it's reasonable/sensible to move this logic elsewhere: it would require passing refs through multiple layers of components, and adding the logic separately for each chart type.

Reply:
refactored to make this hopefully all a lot clearer